stackslib/src/chainstate/stacks/db/mod.rs

```diff
@@ -97,10 +98,24 @@ pub mod headers;
 pub mod transactions;
 pub mod unconfirmed;
 
+#[cfg(not(test))]
 lazy_static! {
     pub static ref TRANSACTION_LOG: bool =
         std::env::var("STACKS_TRANSACTION_LOG") == Ok("1".into());
 }
+#[cfg(not(test))]
+fn is_transaction_log_enabled() -> bool {
+    *TRANSACTION_LOG
+}
+
+#[cfg(test)]
+thread_local! {
+    pub static TRANSACTION_LOG: RefCell<bool> = RefCell::new(false);
```

Comment:
TODO: we need to come up with a better way of doing this.  Ideally it would go into `StacksChainState`'s `DBConfig`, so it would be a setting that gets loaded automatically whenever a `StacksChainState` is instantiated.

Reply:
I still kinda hate the current approach I used, i will invest a bit in checking if adding it to the DBConfig is a viable/quick one before moving the draft to a full pull request

Reply:
Yes, please do. I'm going to insist that this happens prior to merge.

Reply:
Actually, on second thought, we can do this in a separate PR.

Reply:
ok, i will do it in another PR. thanks.